src/signaling/backend_configuration.go

```diff
@@ -148,6 +148,101 @@ func NewBackendConfiguration(config *goconf.ConfigFile) (*BackendConfiguration,
 	}, nil
 }
 
+func (b *BackendConfiguration) RemoveBackend(host string) {
+	delete(b.backends, host)
+}
+
+func (b *BackendConfiguration) UpsertHost(host string, backends []*Backend) {
+	existingIndex := 0
+	for _, existingBackend := range b.backends[host] {
+		found := false
+		index := 0
+		for _, newBackend := range backends {
+			if existingBackend == newBackend {
+				found = true
+				backends = append(backends[:index], backends[index+1:]...)
+				continue
+			}
+			index++
+		}
+		if !found {
+			b.backends[host] = append(b.backends[host][:existingIndex], b.backends[host][existingIndex+1:]...)
+		}
+		existingIndex++
+	}
+
+	b.backends[host] = append(b.backends[host], backends...)
+}
+
+func (b *BackendConfiguration) Reload(config *goconf.ConfigFile) {
+	type configuredBackend struct {
+	}
+	if backendIds, _ := config.GetString("backend", "backends"); backendIds != "" {
```

Comment:
This shares a lot of code with https://github.com/strukturag/nextcloud-spreed-signaling/blob/8a297b1a7067367b29aa070138f48db1620250a6/src/signaling/backend_configuration.go#L72:L107

Could you please move the common option handling / parsing code to a shared function?